utils/kubernetes/describe/describe_test.go

```diff
@@ -0,0 +1,196 @@
+package describe
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+	
+)
+
+// A MockClient meant to interact with the Kubernetes Api
+type MockClient struct {
+	Response      []byte //stores the response body that should be returned when the MockClient makes a request.
+	RequestUrl    string // stores the URL that the Mock client should expect when the HTTP request is performed
+	RequestMethod string // stores the HTTP request method made by the mock client
+	RequestBody   []byte // stores the request body that the fake client should expect to receive when the HTTP request is made
+	DescribeError error  //handle errors for Describe()
+
+}
+
+// the main role of the Prepend Reactor is to return httpErrors for Describerfunc
+// PrependReactor adds a reactor function that gets called before every mock request is sent.
+// The reactor function sets the HTTP status code to 404 if the Describe() function is called.
+func (m *MockClient) PrependReactor(method string, Name string, reactorFunc func(*http.Request) (*http.Response, error)) {
+	originalDo := m.Do
+
+	_ = func(req http.Request) (*http.Response, error) {
+		//call the reactor funtion for every reqeust
+		resp, err := reactorFunc(&req)
+		if err != nil || resp != nil {
+			// If the reactor function returns a response or error, return it immediately
+			return resp, err
+		}
+		// Otherwise, call the original Do() method
+		return originalDo(req)
+	}
+}
+
+// Do executes request and returns a response
+func (m *MockClient) Do(req http.Request) (*http.Response, error) {
+	m.RequestUrl = req.URL.String()
+	m.RequestMethod = req.Method
+	m.RequestBody, _ = ioutil.ReadAll(req.Body)
+
+	if m.DescribeError != nil && strings.Contains(m.RequestUrl, "describe") {
+		return &http.Response{
+			StatusCode: 404,
+			Body:       io.NopCloser(bytes.NewReader(m.Response)),
+		}, m.DescribeError
+	}
+	return &http.Response{
+		StatusCode: 404,
+		Body:       io.NopCloser(bytes.NewReader(m.Response)),
+	}, m.DescribeError
```

Comment:
if & else case seem the same

Reply:
Also, I don't think `io.NopCloser` wrap is required as `bytes.Reader` already satisfies the `io.ReadCloser` interface.

Reply:
ok i see ,  i would change this